### PR TITLE
[CURA-1445 + CURA-1615] Changes on SliceInfo and GCodeWriter

### PR DESCRIPTION
**Pull request by thopiekar**
I think this is the way YouMagine expects the data to be sent.

### CODE REVIEW

**Comment by thopiekar** [comment]
Depends on https://github.com/Ultimaker/Uranium/pull/152

**Comment by thopiekar** [comment]
The changes on the GCodeWriter are more related to CURA-1615..

File: plugins/GCodeWriter/GCodeWriter.py [hunk 1]
```diff
@@ -69,8 +70,13 @@ def _serialiseSettings(self, settings):
         prefix_length = len(prefix)
 
         global_stack = Application.getInstance().getGlobalContainerStack()
-        container_with_profile = global_stack.findContainer({"type": "quality"})
-        serialized = container_with_profile.serialize()
+        # Collecting all data from the top of the stack
+        temporarySettings= InstanceContainer("")
+        temporarySettings.setDefinition(ContainerRegistry.getInstance().findDefinitionContainers(id="fdmprinter")[0])
+        for key in global_stack.getAllKeys():
```
Comment (nallath) [inline plugins/GCodeWriter/GCodeWriter.py]: This does not get all settings; This only gets settings for single extrusion machines.

File: plugins/SliceInfoPlugin/SliceInfo.py [hunk 2]
```diff
@@ -88,7 +88,7 @@ def _onWriteStarted(self, output_device):
                 "settings": global_container_stack.serialize(), # global_container with references on used containers
                 "version": Application.getInstance().getVersion(),
                 "modelhash": "None",
-                "printtime": print_information.currentPrintTime.getDisplayString(),
+                "printtime": print_information.currentPrintTime.getDurationInISOFormat(),
```
Comment (nallath) [inline plugins/SliceInfoPlugin/SliceInfo.py]: I might be wrong, but i think they just want print time in seconds.

Comment (thopiekar) [inline plugins/SliceInfoPlugin/SliceInfo.py]: I think we should wait until we get feedback from YouMagine to be sure :wink:

File: plugins/GCodeWriter/GCodeWriter.py (hunk 1 above)
Comment (thopiekar) [inline plugins/GCodeWriter/GCodeWriter.py]: Can you give a small hint how to get per extruder settings?

Comment (Ghostkeeper) [inline plugins/GCodeWriter/GCodeWriter.py]: `global_stack`.getAllKeys() gets all keys in FDMPrinter. You'd have to call this function on an extruder stack to get the keys in FDMExtruder. I believe those stacks are held in Cura/Settings/ExtruderManager.py.

**Pull request closed without merging**